src/components/TodoHeader.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { ErrorMessage } from '../types/Error';
+
+type HeaderProps = {
+  todos: Todo[],
+  addNewTodo: (todoTitle: string) => Promise<void>,
+  setErrorMessage: (errorMessage: ErrorMessage) => void,
+};
+
+export const Header: React.FC<HeaderProps> = ({
+  todos,
+  addNewTodo,
+  setErrorMessage,
+}) => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+
+  const allTodosCompleted = todos.every(todo => todo.completed);
+
+  const handleFormSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle.trim().length) {
+      setErrorMessage(ErrorMessage.TITLE);
+      setTodoTitle('');
```

Comment:
I think it's redundant, this code only gets executed when title is already an empty string
```suggestion
      ```